cfgov/unprocessed/apps/filing-instruction-guide/js/fig-sidenav.js

```diff
@@ -1,68 +1,121 @@
-    if ( i === len || sY < offsets[i] ) {
-      const hl = i ? i - 1 : i;
-      if ( hl === lastTargetIndex ) return;
-
-      if ( lastTargetIndex !== undefined ) {
-        tocHeaders[lastTargetIndex].classList.remove( 'm-nav-link__current' );
-        getParentHeader( lastTargetIndex ).classList.remove( 'parent-header' );
-      }
+ * Update the sidenav by highlighting the current section and showing
+ * parent nav items as needed
+ * @param {string} target - `href` value of the nav item to highlight
+ */
+const updateNav = target => {
+  if ( target === prevTarget ) return;
+
+  // Highlight the current nav item and expand its container if necessary
+  const targetContainer = getNavItemContainer( target );
+  showElement( targetContainer );
+  highlightNavItem( target );
+  scrollNavItem( target );
+
+  // Remove the highlighting from the previous menu item and close its container
+  if ( prevTarget ) {
+    unHighlightNavItem( prevTarget );
+    const prevTargetContainer = getNavItemContainer( prevTarget );
+    if ( prevTargetContainer !== targetContainer ) {
+      hideElement( prevTargetContainer );
+    }
+  }
+
+  prevTarget = target;
+};
 
-      tocHeaders[hl].classList.add( 'm-nav-link__current' );
-      getParentHeader( hl ).classList.add( 'parent-header' );
-      lastTargetIndex = hl;
-      return;
+/**
+ * Callback for IntersectionObserver
+ * @param {IntersectionObserverEntry} entries - array of observer entries
+ * See https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserverEntry
+ */
+const handleIntersect = entries => {
+  entries.forEach( entry => {
+    if ( entry.intersectionRatio > 0 ) {
+      updateNav( entry.target.getAttribute( 'href' ) );
     }
+  } );
+};
+
+/**
+ * init - Initialize everything on page load.
+ */
+const init = () => {
+  /* Only proceed if IntersectionObserver is supported (everything except IE)
+     and we're on a larger screen
+     See https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API */
+  if ( 'IntersectionObserver' in window && viewportIsIn( DESKTOP ) ) {
+    appRoot().querySelectorAll( '.o-secondary-navigation_list__children' ).forEach( ul => {
+      hideElement( ul );
+    } );
+
+    const observer = new IntersectionObserver( handleIntersect, {
+      root: document,
+
+      /* Sets an intersection area that spans 5% above the top of the viewport and
+         95% above the bottom of the viewport, resulting in a box that is 10% of
+         the viewport height with 5% hanging over the top. */
+      rootMargin: '5% 0px -95% 0px'
+    } );
+
+    const sections = appRoot().querySelectorAll( 'a[data-scrollspy]' );
+
+    // Highlight the first section on page load
+    updateNav( sections[0].getAttribute( 'href' ) );
+
+    sections.forEach( section => observer.observe( section ) );
+  }
+
+  if ( !viewportIsIn( DESKTOP ) ) {
+    appRoot().addEventListener( 'click', event => {
+      if ( event.target.classList.contains( 'm-nav-link' ) ) {
```

Comment:
I think this should also target `.o-fig_heading > a` elements so that clicking on the headings in the text also utilizes `scrollIntoViewWithOffset` and doesn't clip the heading text behind the sticky nav